sn_networking/src/external_address.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+#![allow(dead_code)]
+
+use crate::{driver::NodeBehaviour, mulitaddr_get_ip};
+use itertools::Itertools;
+use libp2p::{Multiaddr, Swarm};
+use std::{collections::HashMap, net::IpAddr};
+
+const MAX_REPORTS_BEFORE_CONFIRMATION: u8 = 3;
+const MAX_CANDIDATES: usize = 50;
+
+/// Manages the external addresses of a Public node. For a relayed node, the RelayManager should deal with
+/// adding and removing external addresses.
+// TODO:
+// 1. if the max candidate is reached, kick out the oldest candidate sorted by # of reports
+#[derive(Debug)]
+pub struct ExternalAddressManager {
+    /// All the external addresses of the node
+    address_states: Vec<ExternalAddressState>,
+    current_ip_address: Option<IpAddr>,
+
+    /// The maximum number of reports before an candidate address is confirmed
+    pub max_reports_before_confirmation: u8,
+    /// The maximum number of candidates to store
+    pub max_candidates: usize,
+}
+
+impl ExternalAddressManager {
+    pub fn new() -> Self {
+        Self {
+            address_states: Vec::new(),
+            current_ip_address: None,
+            max_reports_before_confirmation: MAX_REPORTS_BEFORE_CONFIRMATION,
+            max_candidates: MAX_CANDIDATES,
+        }
+    }
+
+    pub fn set_reports_before_confirmation(&mut self, max_reports_before_confirmation: u8) {
+        self.max_reports_before_confirmation = max_reports_before_confirmation;
+    }
+
+    pub fn set_max_candidates(&mut self, max_candidates: usize) {
+        self.max_candidates = max_candidates;
+    }
+
+    /// Get the list of candidate addresses
+    pub fn get_candidate_addresses(&self) -> Vec<&Multiaddr> {
```

Comment:
Again, no `get_` (https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter)

```suggestion
    pub fn candidates(&self) -> Vec<&Multiaddr> {
```